cargo-miri/src/util.rs

```diff
@@ -249,3 +251,26 @@ pub fn debug_cmd(prefix: &str, verbose: usize, cmd: &Command) {
     }
     eprintln!("{prefix} running command: {cmd:?}");
 }
+
+/// Determines where the sysroot of this exeuction is
+/// 
+/// Either in a user-specified spot by an envar, or in a default cache location.
+pub fn get_sysroot() -> PathBuf {
+    match std::env::var_os("MIRI_SYSROOT") {
+        Some(dir) => PathBuf::from(dir),
+        None => {
+            let user_dirs = directories::ProjectDirs::from("org", "rust-lang", "miri").unwrap();
+            user_dirs.cache_dir().to_owned()
+        }
+    }
+}
+
+/// Deletes the Miri sysroot cache directory.
+pub fn clean_sysroot() {
+    let sysroot_dir = get_sysroot();
+
+    eprint!("[cargo-miri] clearing cache at {}", sysroot_dir.display());
+
+    // Keep it simple
+    std::fs::remove_dir_all(sysroot_dir).unwrap();
```

Comment:
I wonder if we should refuse to do this when MIRI_SYSROOT is set. Otherwise there's a risk that it is set incorrect and then we'll delete the wrong thing...

Reply:
Yeah that's reasonable.